libs/blocks/global-navigation/utilities/utilities.js

```diff
@@ -243,3 +288,46 @@ export function processMartechAttributeMetadata(html) {
   });
   return null;
 }
+
+export async function fetchAndProcess({ url, message, shouldDecorateLinks = true } = {}) {
+  try {
+    const path = getFederatedUrl(url);
+    const res = await fetch(path.replace(/(\.html$|$)/, '.plain.html'));
+    const text = await res.text();
+    const { body } = new DOMParser().parseFromString(text, 'text/html');
+
+    const inlineFrags = [...body.querySelectorAll('a[href*="#_inline"]')];
+    if (inlineFrags.length) {
+      const { default: loadInlineFrags } = await import('../../fragment/fragment.js');
+
+      const fragPromises = inlineFrags.map((link) => {
+        // Replacing paragraphs should happen in the fragment module
+        // https://jira.corp.adobe.com/browse/MWPW-141039
+        if (link.parentElement && link.parentElement.nodeName === 'P') {
+          const div = document.createElement('div');
+          link.parentElement.replaceWith(div);
+          div.appendChild(link);
+        }
+        link.href = getFederatedUrl(link.href);
+        return loadInlineFrags(link);
+      });
+      await Promise.all(fragPromises);
+    }
+
+    if (path.includes('/federal/')) federatePictureSources(body);
+
+    if (shouldDecorateLinks) decorateLinks(body);
+
+    const blocks = body.querySelectorAll('.martech-metadata');
+    blocks.forEach((block) => {
+      import('../../martech-metadata/martech-metadata.js')
+        .then(({ default: decorate }) => decorate(block));
+    });
```

Comment:
Could we not check for the `.martech-metadata` class and if there are any matches, import the module, which, when resolved, decorates all the blocks? The resource will likely be cached, but I think it's an easy enough change to avoid eventual multiple requests altogether.

Reply:
Copy/pasted this from the base work Vivian had done, I refactored this slightly